### DISCUSSION TITLE
maths on active scalar

### DISCUSSION THREAD

**Discussion by craigmillernz** [pyvista/pyvista#2603] [Q&A]
I'd like to do maths on the active scalar, such as taking the log10.
`np.log10(model.active_scalar) ` returns a numpy array not a vtk object.
Alternatively can I log transform the colour scale?

**Comment by banesullivan** [comment]
Hi @craigmillernz - what you want to do is totally possible! PyVista is built for direct use with NumPy so there's no need to worry about VTK data types and all of those nuances. As a user, all you have to do is pass a numpy array to your PyVista mesh and PyVista will handle the rest.

Here's an example:

```py
>>> import pyvista as pv
>>> import numpy as np
>>> from pyvista import examples
>>> mesh = examples.download_st_helens().warp_by_scalar()
>>> mesh.plot(use_panel=False)
```

![download](https://user-images.githubusercontent.com/22067021/58678367-bc1f4680-831c-11e9-8afb-e0a9cc53a313.png)


Perform some math like `np.log10`
```py
>>> myarr = np.log10(mesh.active_scalar)
>>> type(myarr)
numpy.ndarray
```

And send that array back to PyVista for plotting
```py
>>> mesh['stuff'] = myarr
>>> mesh.plot(scalars='stuff', use_panel=False)
```

![download](https://user-images.githubusercontent.com/22067021/58678375-c3465480-831c-11e9-9b8c-ef19a275ec9b.png)



And a sanity check:
```py
>>> np.allclose(mesh['stuff'], np.log10(mesh.active_scalar))
True
```

**Comment by banesullivan** [comment]
> Alternatively can I log transform the colour scale?

As for transforming the color mapping... this is still a work in progress that I haven't figured out quite yet. A `norm` option was requested in https://github.com/pyvista/pyvista/issues/122

**Comment by craigmillernz** [comment]
Thanks, it was getting it back into pyvista that was the trick.
